--- a/classes/data/File.class.php
+++ b/classes/data/File.class.php
@@ -450,7 +450,7 @@
 
         // add natural, human sorting, in both databases
         if ($dbtype == 'pgsql') {
-            $sql .= " SUBSTRING(name FROM '^[A-Za-z]+') , CAST(SUBSTRING(name FROM '\d+') AS BIGINT) ";
+            $sql .= " SUBSTRING(name FROM '^[A-Za-z]+') , CAST(SUBSTRING(name FROM '\d+') AS NUMERIC) ";
         }
         if ($dbtype == 'mysql') {
             $sql .= " NATURAL_SORT_KEY(name) ";
